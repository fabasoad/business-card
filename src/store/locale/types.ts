export const LOAD_LOCALE = 'LOAD_LOCALE'
export const SET_LOCALE = 'SET_LOCALE'

export class Locale {
  code: string
  title: string

  constructor(code: string, title: string = null) {
    this.code = code
    this.title = ` ${title || code}`
  }
}

interface LoadLocaleAction {
  type: typeof LOAD_LOCALE
  payload: Locale
}

interface SetLocaleAction {
  type: typeof SET_LOCALE
  payload: Locale
}

export type LocaleActionTypes = LoadLocaleAction | SetLocaleAction

export type AppActions = LocaleActionTypes
