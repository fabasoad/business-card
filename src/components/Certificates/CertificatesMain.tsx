import * as React from 'react'
import { useTranslation } from 'react-i18next'
import CertificateItem from './CertificateItem'

import * as items from './items.json'
import CertificateIssuerStorage
  from '../../scripts/certificates/CertificateIssuerStorage'
import TechnologyStorage from '../../scripts/technologies/TechnologyStorage'

interface CertificatesMainProps {
  certificateIssuerStorage?: CertificateIssuerStorage
  technologyStorage?: TechnologyStorage
}

export default function CertificatesMain(props: CertificatesMainProps) {
  const { t } = useTranslation()
  return (
    <div id="certificates" className="light-component">
      <div className="section-title text-center center">
        <h2>{t('business-card-certificates-title')}</h2>
        <hr />
      </div>
      <div className="container">
        <div className="row justify-content-center">
          {items
            .sort((a, b) => -a.date.localeCompare(b.date))
            .map((item) => <CertificateItem
              key={item.id}
              id={item.id}
              issueDate={new Date(item.date)}
              issuer={props.certificateIssuerStorage.findByName(item.issuer)}
              technology={props.technologyStorage.findByName(item.technology)}
              i18nTitleKey={item.i18nTitleKey}
              url={item.url}
            />)}
        </div>
      </div>
    </div>
  )
}

CertificatesMain.defaultProps = {
  certificateIssuerStorage: new CertificateIssuerStorage(),
  technologyStorage: new TechnologyStorage()
}
