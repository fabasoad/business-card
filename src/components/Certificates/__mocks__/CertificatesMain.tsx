import * as React from 'react'

export default function CertificatesMain() {
  return (
    <div data-testid="CertificatesMain"></div>
  )
}
