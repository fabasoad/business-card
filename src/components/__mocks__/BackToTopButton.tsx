import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
