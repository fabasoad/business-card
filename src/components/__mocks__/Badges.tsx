import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
