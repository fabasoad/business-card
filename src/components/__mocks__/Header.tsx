import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
