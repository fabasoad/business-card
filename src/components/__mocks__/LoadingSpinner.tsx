import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
