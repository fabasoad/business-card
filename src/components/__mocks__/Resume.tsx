import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
