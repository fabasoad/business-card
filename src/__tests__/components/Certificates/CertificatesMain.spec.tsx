/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import CertificatesMain from '../../../components/Certificates/CertificatesMain'
import { useTranslation } from '../../__mocks__/react-i18next'
import CertificateIssuerStorage from '../../../scripts/certificates/CertificateIssuerStorage'
import TechnologyStorage from '../../../scripts/technologies/TechnologyStorage'
import { CertificateIssuer } from '../../../scripts/certificates/types'
import { Technology } from '../../../scripts/technologies/types'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render CertificatesMain correctly', () => {
  const cisFindByNameSpy = jest.fn((k) => ({ name: `test-name-${k}`, img: `test-img-${k}` }))
  const cis = new CertificateIssuerStorage({ items: new Map<string, CertificateIssuer>() })
  cis.findByName = cisFindByNameSpy

  const tsFindByNameSpy = jest.fn((k) => ({ name: `test-name-${k}`, title: `test-title-${k}`, img: `test-img-${k}` }))
  const ts = new TechnologyStorage({ items: new Map<string, Technology>() })
  ts.findByName = tsFindByNameSpy

  const wrapper: ShallowWrapper = shallow(<CertificatesMain
    certificateIssuerStorage={cis}
    technologyStorage={ts}
  />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(1)
  expect(tMock).toHaveBeenCalledWith('business-card-certificates-title')
  expect(cisFindByNameSpy).toHaveBeenCalledTimes(4)
  new Array<string>('coursera', 'oracle', 'udemy').forEach((t) =>
    expect(cisFindByNameSpy).toHaveBeenCalledWith(t)
  )
  expect(tsFindByNameSpy).toHaveBeenCalledTimes(4)
  new Array<string>('python', 'java', 'react').forEach((t) =>
    expect(tsFindByNameSpy).toHaveBeenCalledWith(t)
  )
})
