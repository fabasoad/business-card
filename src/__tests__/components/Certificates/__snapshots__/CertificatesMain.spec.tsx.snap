// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CertificatesMain correctly 1`] = `
<div
  className="light-component"
  id="certificates"
>
  <div
    className="section-title text-center center"
  >
    <h2>
      business-card-certificates-title
    </h2>
    <hr />
  </div>
  <div
    className="container"
  >
    <div
      className="row justify-content-center"
    >
      <CertificateItem
        i18nTitleKey="business-card-certificate-4"
        id=""
        issueDate={2020-05-01T00:00:00.000Z}
        issuer={
          Object {
            "img": "test-img-oracle",
            "name": "test-name-oracle",
          }
        }
        key=""
        technology={
          Object {
            "img": "test-img-java",
            "name": "test-name-java",
            "title": "test-title-java",
          }
        }
        url="https://www.youracclaim.com/badges/689b9d7e-cae5-4e36-8a1f-c8c34f104249/public_url"
      />
      <CertificateItem
        i18nTitleKey="business-card-certificate-3"
        id="UC-JGDVYKAQ"
        issueDate={2019-10-01T00:00:00.000Z}
        issuer={
          Object {
            "img": "test-img-udemy",
            "name": "test-name-udemy",
          }
        }
        key="UC-JGDVYKAQ"
        technology={
          Object {
            "img": "test-img-react",
            "name": "test-name-react",
            "title": "test-title-react",
          }
        }
        url="http://ude.my/UC-JGDVYKAQ"
      />
      <CertificateItem
        i18nTitleKey="business-card-certificate-2"
        id="W5AB2PBZQNPY"
        issueDate={2018-03-01T00:00:00.000Z}
        issuer={
          Object {
            "img": "test-img-coursera",
            "name": "test-name-coursera",
          }
        }
        key="W5AB2PBZQNPY"
        technology={
          Object {
            "img": "test-img-java",
            "name": "test-name-java",
            "title": "test-title-java",
          }
        }
        url="https://www.coursera.org/account/accomplishments/specialization/W5AB2PBZQNPY"
      />
      <CertificateItem
        i18nTitleKey="business-card-certificate-1"
        id="MWE43H8YDS"
        issueDate={2015-05-01T00:00:00.000Z}
        issuer={
          Object {
            "img": "test-img-coursera",
            "name": "test-name-coursera",
          }
        }
        key="MWE43H8YDS"
        technology={
          Object {
            "img": "test-img-python",
            "name": "test-name-python",
            "title": "test-title-python",
          }
        }
        url="https://www.coursera.org/account/accomplishments/verify/MWE43H8YDS"
      />
    </div>
  </div>
</div>
`;
